airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -385,3 +399,45 @@ def state(self) -> MutableMapping[str, Any]:
     def state(self, value: StreamState):
         """State setter, accept state serialized by state getter."""
         self.stream_slicer.update_cursor(value)
+
+    def parse_records_and_emit_request_and_responses(self, request, response, stream_slice, stream_state) -> Iterable[AirbyteMessage]:
+        yield self._create_trace_message_from_request(request)
+        yield self._create_trace_message_from_response(response)
+        # Not great to need to call _read_pages which is a private method
+        # A better approach would be to extract the HTTP client from the HttpStream and call it directly from the HttpRequester
+        for record_mapping in self._read_pages(
+            lambda req, res, state, _slice: self.parse_response(res, stream_slice=_slice, stream_state=state), stream_slice, stream_state
+        ):
+            # FIXME: need to get the real json schema if we want to support type transforms
+            json_schema = {}
```

Comment:
deleted as discussed offline. the gist of the rational was:
> AirbyteMessage used to be created by the AbstractSource, but Streams can also create them now.
> From the Stream's perspective, I think we can either
> 1. return a dict, an AirbyteLogMessage, or an AirbyteTraceMessage, but never an AirbyteRecordMessage (because the transformation is needed to create the message)
> 2. create the AirbyteRecordMessage without the transformer and ask the AbstractSource to apply the transforms if needed
> 3. let the stream be aware of the transformer and schema so they can apply the transforms

we decided to go with 1 because it minimizes confusion